services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/utils/DataPlaneManagerUtil.java

```diff
@@ -0,0 +1,704 @@
+                                                     int hostNum,
+                                                     int subnetNum,
+                                                     int NumOfIPsInSubnet1,
+                                                     int NumOfIPsInSubnet2,
+                                                     boolean hasInternalRouterInfo,
+                                                     boolean hasInternalSubnetRoutingTable,
+                                                     boolean hasInternalRoutingRule,
+                                                     boolean hasNeighbor,
+                                                     int neighborNum,
+                                                     boolean fastPath) {
+        NetworkConfiguration networkConfiguration = new NetworkConfiguration();
+
+        // set operationType and resourceType
+        networkConfiguration.setOpType(Common.OperationType.forNumber(operationType));
+        networkConfiguration.setRsType(Common.ResourceType.forNumber(resourceType));
+
+        // set routers_internal
+        List<InternalRouterInfo> internalRouterInfos = new ArrayList<>();
+        if (hasInternalRouterInfo) {
+            InternalRouterInfo routerInfo = new InternalRouterInfo();
+            InternalRouterConfiguration internalRouterConfiguration = new InternalRouterConfiguration();
+            List<InternalSubnetRoutingTable> subnet_routing_tables = new ArrayList<>();
+
+            if (hasInternalSubnetRoutingTable) {
+                InternalSubnetRoutingTable internalSubnetRoutingTable = new InternalSubnetRoutingTable();
+                List<InternalRoutingRule> routing_rules = new ArrayList<>();
+
+                if (hasInternalRoutingRule){
+                    InternalRoutingRule internalRoutingRule = new InternalRoutingRule();
+                    routing_rules.add(internalRoutingRule);
+                }
+
+                internalSubnetRoutingTable.setRoutingRules(routing_rules);
+                subnet_routing_tables.add(internalSubnetRoutingTable);
+            }
+
+            internalRouterConfiguration.setSubnetRoutingTables(subnet_routing_tables);
+
+
+            //routerInfo.setOperationType(OperationType.valueOf("create"));
+            routerInfo.setRouterConfiguration(internalRouterConfiguration);
+            internalRouterInfos.add(routerInfo);
+        }
+
+        networkConfiguration.setInternalRouterInfos(internalRouterInfos);
+
+        // set neighborInfos
+        List<NeighborInfo> neighborINFO = new ArrayList<>();
+        if (hasNeighbor) {
+            for (int i = 0; i < neighborNum; i ++) {
+                NeighborInfo neighborInfo = new NeighborInfo(DPMAutoUnitTestConstant.hostIp + i, DPMAutoUnitTestConstant.hostId, DPMAutoUnitTestConstant.portId + i, DPMAutoUnitTestConstant.portMac, DPMAutoUnitTestConstant.portIp, DPMAutoUnitTestConstant.vpcId, DPMAutoUnitTestConstant.subnetId + i);
```

Comment:
If we use DPMAutoUnitTestConstant.portMac and DPMAutoUnitTestConstant.portIp, do that mean that we are generating multiple neighbor ports with same IP and MAC?

Reply:
Yes, because we assume there is one neighbor if it has neighbor in most of cases. Do we have the case to test multiple neighbor with different IP and MAC?